lom/Makefile

```diff
@@ -61,97 +64,164 @@ format_code:
 
 # Define the file coverage extraction function
 define extract_file_coverage
-	@echo "*******************************************************"
-	@echo "Extracting file coverages:"
-	@file_coverage="$$(sed -n 's/.*<option value="[^"]\+">\([^<]\+\).* (\([^)]\+\%.*\)<\/option>.*/\1 (\2)/p' $(1))"; 
-	echo "$${file_coverage}"
-	@echo "*******************************************************"
-	@echo ""
-	@echo ""
+#!/bin/bash
+echo "====================================================================================="
+echo -e "Detail file coverage from $$1 :"
+echo "_____________________________________________________________________________________"
+file_coverage="$$(sed -n 's/.*<option value="[^"]\+">\([^<]\+\).* (\([^)]\+\%.*\)<\/option>.*/\1 (\2)/p' "$$1")"; 
+echo "$${file_coverage}"
+echo "====================================================================================="
 endef
+export extract_file_coverage
+
+add-extract_file_coverage: 
+	$(shell mkdir -p $(BUILD_DIR))
+	$(shell rm -f  $(BUILD_DIR)/extract_file_coverage.sh)
+	$(shell echo >> $(BUILD_DIR)/extract_file_coverage.sh)
+	@echo "$$extract_file_coverage" >> $(BUILD_DIR)/extract_file_coverage.sh
+	$(shell chmod 777 $(BUILD_DIR)/extract_file_coverage.sh 2>/dev/null || :)
```

Comment:
1.what is the purpose of ||
2. Why  do we need to redirect to stderr to null

Reply:
removed null redirect